packages/dd-trace/src/appsec/api_security_sampler.js

```diff
@@ -1,61 +1,47 @@
 'use strict'
 
-const log = require('../log')
+const ApiSecuritySamplerCache = require('./api_security_sampler_cache')
+const PrioritySampler = require('../priority_sampler')
+const web = require('../plugins/util/web')
 
 let enabled
-let requestSampling
-
-const sampledRequests = new WeakSet()
+let sampledRequests
+const prioritySampler = new PrioritySampler()
 
 function configure ({ apiSecurity }) {
   enabled = apiSecurity.enabled
-  setRequestSampling(apiSecurity.requestSampling)
+  sampledRequests = new ApiSecuritySamplerCache(apiSecurity.sampleDelay)
 }
 
 function disable () {
   enabled = false
+  sampledRequests?.clear()
 }
 
-function setRequestSampling (sampling) {
-  requestSampling = parseRequestSampling(sampling)
-}
-
-function parseRequestSampling (requestSampling) {
-  let parsed = parseFloat(requestSampling)
-
-  if (isNaN(parsed)) {
-    log.warn(`Incorrect API Security request sampling value: ${requestSampling}`)
+function sampleRequest (req, res) {
+  if (!enabled) return false
 
-    parsed = 0
-  } else {
-    parsed = Math.min(1, Math.max(0, parsed))
-  }
+  const rootSpan = web.root(req)
+  if (!rootSpan) return false
 
-  return parsed
-}
+  const isSampled = prioritySampler.isSampled(rootSpan)
```

Comment:
This method set the priority sampler in the trace. Should we set the trace priority if, at the end, it was already sampled? Should we first check in our map, or in the trace?

imho, we should check first our map, and once that we have checked that we want to send it to the waf, check if the trace will be maintained or not.
if we want to maintain and prioirtySampler says that it is OK, update our map.

wdyt?
cc @iunanua

Reply:
it was discussed. I guess we can resolve the conversation.